packages/light.js/src/frequency/utils/createPubsubObservable.ts

```diff
@@ -12,14 +12,36 @@ import { switchMap } from 'rxjs/operators';
 import { createApiFromProvider, getApi } from '../../api';
 import { distinctReplayRefCount } from '../../utils/operators/distinctReplayRefCount';
 
+const POLL_INTERVAL = 1000;
+
+/**
+ * Create a polling function.
+ *
+ * @ignore
+ */
+function createPoll<T> (
+  fallback: string,
+  api: any,
+  pollInterval = POLL_INTERVAL
+) {
+  const [fallbackNamespace, fallbackMethod] = fallback.split('_');
+
+  return timer(0, pollInterval).pipe(
+    switchMap(() => api[fallbackNamespace][fallbackMethod]())
```

Comment:
might not be a problem but for RPC calls taking >1 second to answer, the observable would never fire

Reply:
I think exhaustMap should do the trick https://runkit.com/axelchalon/5c00004a6d959e0012c60bc3 

We have a timer (source observable) ; we make the RPC query each time the timer fires, unless the previous RPC hasn't resolved yet